draft-ietf-quic-http.md

```diff
@@ -508,113 +508,110 @@ HEADERS frames can only be sent on request / push streams.
 The PRIORITY (type=0x2) frame specifies the client-advised priority of a
 request, server push or placeholder.
 
-A PRIORITY frame identifies an element to prioritize, and an element upon which
-it depends.  A Prioritized ID or Dependency ID identifies a client-initiated
-request using the corresponding stream ID, a server push using a Push ID (see
-{{frame-push-promise}}), or a placeholder using a Placeholder ID (see
-{{placeholders}}).
+A PRIORITY frame identifies an element to prioritize and a placeholder upon
+which it depends, or the root if no placeholder is specified.  A Prioritized
+ID identifies a server push using a Push ID (see {{frame-push-promise}}),
+or a placeholder using a Placeholder ID (see {{placeholders}}).
 
 When a client initiates a request, a PRIORITY frame MAY be sent as the first
 frame of the stream, creating a dependency on an existing element.  In order to
 ensure that prioritization is processed in a consistent order, any subsequent
-PRIORITY frames for that request MUST be sent on the control stream.  A
-PRIORITY frame received after other frames on a request stream MUST be treated
-as a connection error of type HTTP_UNEXPECTED_FRAME.
+PRIORITY frames for that request MUST be sent on the request stream.
```

Comment:
The difficulty here is that you'll need to keep the request stream open (from the client-send side) for as long you want to reprioritize the request. This risks conflicting with the current text in section 5.1 

> An HTTP request/response exchange fully consumes a bidirectional QUIC
   stream.  After sending a request, a client MUST close the stream for
   sending.

That is not insurmountable, you just might need some extra text in the draft to deal with the ripple effects of the changes.

Reply:
I've now reverted this portion of the change, thanks for the suggestion.